File: PalindromeNumber/PalindromeNumber.py
def isPalindrome(x: int) -> bool:
    return str(x) == str(x)[::-1]

if __name__ == "__main__":
    print(isPalindrome(121))
    print(isPalindrome(-121))
    print(isPalindrome(10))
    print(isPalindrome(-101))
    print(isPalindrome(0))
    print(isPalindrome(11))
    

File: PalindromeNumber/PalindromeNumber.cpp
#include<iostream>
using namespace std;

bool isPalindrome(int x) {
    if (x < 0)  return false;

    long reversed = 0;
    long temp = x;

    while(temp != 0){
        int digit = temp % 10;
        reversed = reversed * 10 + digit;
        temp /= 10;
    }

    return (reversed == x);
}

int main(){
    int x = 121;
    cout << isPalindrome(x) << endl;
    return 0;
}